### ISSUE TITLE
Mono projects fail to build without Godot_DEBUG_MSBUILD defined

### ISSUE THREAD

**Issue by KellyThomas** [godotengine/godot#19394]
**Godot version:**
master 07d4cd7f6

**OS/device including version:**
Windows 7

**Issue description:**
My projects are building successfully when I have an environment variable set to either:
`Godot_DEBUG_MSBUILD=0` or
`Godot_DEBUG_MSBUILD=1`

But if `Godot_DEBUG_MSBUILD` is undefined then my build fails with this message in my console window:
````
Mono: Initializing module...
Mono: Runtime initialized
Mono: INITIALIZED
ERROR: CSharpScript::instance_create: Condition ' !valid ' is true. returned: 0
   At: modules\mono\csharp_script.cpp:1969

Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at GodotSharpTools.Build.BuildInstance.IsDebugMSBuildRequested () [0x0000a] in <99a4e44fff2b4c7c835bbaf5a480f241>:0
  at GodotSharpTools.Build.BuildInstance.Build (System.String loggerAssemblyPath, System.String loggerOutputDir, System.
String[] customProperties) [0x00000] in <99a4e44fff2b4c7c835bbaf5a480f241>:0
ERROR: The build method threw an exception.
System.NullReferenceException: Object reference not set to an instance of an object
   At: modules\mono\editor\godotsharp_builds.cpp:547
````

**Comment by KellyThomas** [comment]
May be related to https://github.com/godotengine/godot/pull/19387

**Comment by neikeq** [comment]
Fixed by #19396
